Remove upper bound dependency limits from gemspec

rafaelfranca opened the pull request:
All that was needed to support Rails 7 in this gem was to update the
gemspec. This means that even in the most likely release to break
anything, a major release, we didn't needed to worry about breaking this
gem.

This upper bound is just creating unnecessary work for the
maintainers of this gem, which would need to release it without any code
change, and the users, which need to open issues asking for new releases.

Let's allow our users to try any version of our dependencies and report
real issues if they find them, and save everyone's time just removing
the upper bound.

javierjulio reviewed (commented):
Thanks @rafaelfranca! I understand the reasoning and am ok with either removing or specifying upper bounds. Why did you remove the `gemspec_lint` test? That would help us confirm no warnings. Other than thought no concerns with the change. There are merge conflicts.

rafaelfranca commented:
> Why did you remove the `gemspec_lint` test? That would help us confirm no warnings

I removed because RubyGems show warnings for dependencies without upper bound. I wish I could just disable that warning so we could checking the others, but that is not the case

deivid-rodriguez commented:
I agree with your reasoning too @rafaelfranca. I would like to work on making RubyGems gemspec warnings configurable in the future. But for now it seems easy enough to ignore this particular warning from the output, instead of removing everything? I'm happy to work on it myself if you don't have the time.

rafaelfranca commented:
You mean in the test? I can totally do it. I'll update the PR to ignore just that warning.

deivid-rodriguez commented:
Yes, in the test I mean.

javierjulio committed:
Remove upper bound limit from our dependencies

All that was needed to support Rails 7 in this gem was to update the
gemspec. This means that even in the most likely release to break
anything, a major release, we didn't needed to worry about breaking this
gem.

This upper bound is just creating unnecessary work for the
maintainers of this gem, which would need to release it without any code
change, and the users, which need to open issues asking for new releases.

Let's allow our users to try any version of our dependencies and report
real issues if they find them, and save everyone's time just removing
the upper bound.

javierjulio committed:
Add changelog entry

javierjulio reviewed (approved):
Thanks!
